sensors/triggers/slack/slack.go

```diff
@@ -114,54 +115,65 @@ func (t *SlackTrigger) Execute(ctx context.Context, events map[string]*v1alpha1.
 
 	api := slack.New(slackToken, slack.OptionDebug(false))
 
-	params := &slack.GetConversationsParameters{
-		Limit: 200,
-		Types: []string{"public_channel", "private_channel"},
-	}
-	channelID := ""
-	isPrivateChannel := false
+	t.Logger.Infow("posting to channel...", zap.Any("channelName", channel))
 	for {
-		channels, nextCursor, err := api.GetConversations(params)
+		channelID, timestamp, err := api.PostMessage(channel, slack.MsgOptionText(message, false))
 		if err != nil {
-			t.Logger.Errorw("unable to list channels", zap.Error(err))
-			return nil, errors.Wrapf(err, "failed to list channels")
-		}
-		for _, c := range channels {
-			if c.Name == channel {
-				channelID = c.ID
-				isPrivateChannel = c.IsPrivate
-				break
+			if err.Error() == "not_in_channel" {
+				isPrivateChannel := false
+				params := &slack.GetConversationsParameters{
+					Limit:           200,
+					Types:           []string{"public_channel", "private_channel"},
+					ExcludeArchived: "true",
+				}
+
+				for {
+					channels, nextCursor, err := api.GetConversations(params)
+					if err != nil {
+						switch e := err.(type) {
+						case *slack.RateLimitedError:
+							<-time.After(e.RetryAfter)
+							continue
+						default:
+							t.Logger.Errorw("unable to list channels", zap.Error(err))
+							return nil, errors.Wrapf(err, "failed to list channels")
+						}
+					}
+					for _, c := range channels {
+						if c.Name == channel {
+							channelID = c.ID
+							isPrivateChannel = c.IsPrivate
+							break
+						}
+					}
+					if nextCursor == "" || channelID != "" {
+						break
+					}
+					params.Cursor = nextCursor
+				}
+				if channelID == "" {
+					return nil, errors.Errorf("failed to get channelID of %s", channel)
+				}
+				if isPrivateChannel {
```

Comment:
I actually not very sure what the logic for private channel should be, but here the logic seems to be changed.

Old: Skip "join conversation", still post message;
New: return an error.

Reply:
The error probably makes sense IMO, because if it's a private channel the bot can't join and can't send a message to it without joining. It needs to be invited.

Reply:
Sounds good.